### PR TITLE
add `endpoint-id` to `az monitor data-collection rule association`

### PR DESCRIPTION
**Pull request by AllyW**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command

az monitor data-collection rule association

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`. 
 closed https://github.com/Azure/azure-cli/issues/26628

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/monitor-control-service/azext_amcs/manual/_params.py [hunk 1]
```diff
@@ -30,6 +32,14 @@
     KnownWindowsEventLogDataSourceStreams, KnownSyslogDataSourceStreams, \
     KnownSyslogDataSourceFacilityNames, KnownSyslogDataSourceLogLevels
 
+logger = get_logger(__name__)
+
+
+def validate_association_name_with_endpoint(cmd, namespace):
+    if namespace.endpoint_id:
+        if namespace.association_name != "configurationAccessEndpoint":
+            raise CLIError("Association name for resource to endpoint must be configurationAccessEndpoint")
```
Comment (kairu-ms) [inline src/monitor-control-service/azext_amcs/manual/_params.py:41]: CLIError is not recommended to use. Please follow this [doc](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md)

Comment (kairu-ms) [inline src/monitor-control-service/azext_amcs/manual/_params.py:41]: This function should be defined in _validators.py file

File: src/monitor-control-service/azext_amcs/manual/_validators.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# --------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+#
+# Code generated by Microsoft (R) AutoRest Code Generator.
+# Changes may cause incorrect behavior and will be lost if the code is
+# regenerated.
```
Comment (kairu-ms) [inline src/monitor-control-service/azext_amcs/manual/_validators.py:8]: You can remove these lines.

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**